Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/organizer/details/QrCodeDetailViewModel.kt

```diff
@@ -29,74 +25,29 @@ class QrCodeDetailViewModel @AssistedInject constructor(
     private val traceLocationRepository: DefaultTraceLocationRepository
 ) : CWAViewModel() {
 
-    private val traceLocationFlow = MutableStateFlow<TraceLocation?>(null)
-    private val titleFlow = MutableStateFlow<String?>(null)
-    private val subtitleFlow = MutableStateFlow<String?>(null)
-    private val startTimeFlow = MutableStateFlow<Instant?>(null)
-    private val endTimeFlow = MutableStateFlow<Instant?>(null)
-    private val bitmapLiveData = MutableLiveData<Bitmap>()
+    private var traceLocation: TraceLocation? = null
```

Comment:
what about lateinit anyway you initialise it in `init` block ?

Reply:
`traceLocation` is initialized inside a coroutine, lateinit wouldn't be safe and could cause a crash.

Reply:
Ok. I had a deeper look , you can keep it nullable but you need to handle an exception that could be thrown by 
` traceLocationRepository.traceLocationForId(traceLocationId)`
inside `launch` 

```kotlin
launch {
            try {
                traceLocation = traceLocationRepository.traceLocationForId(traceLocationId).also {
                    mutableUiState.postValue(UiState(it))
                    createQrCode(it)
                }
            } catch (e: Exception) {
                Timber.d(e, "No location found")
                e.report(ExceptionCategory.INTERNAL)
            }
        }
```

Reply:
good idea, updated